src/software/jetson_nano/services/motor.cpp

```diff
@@ -898,3 +985,90 @@ void MotorService::startController(uint8_t motor, bool dribbler)
         configureEncoder(motor);
     }
 }
+
+void MotorService::checkEncoderConnections()
+{
+    LOG(INFO) << "Starting encoder connection check!";
+
+    std::vector<bool> calibrated_motors(NUM_DRIVE_MOTORS, false);
+    std::vector<int> initial_velocities(NUM_DRIVE_MOTORS, 0);
+
+    for (uint8_t motor = 0; motor < NUM_DRIVE_MOTORS; ++motor)
+    {
+        // read back current velocity
+        initial_velocities[motor] = tmc4671_readInt(motor, TMC4671_ABN_DECODER_COUNT);
+
+        // open loop mode can be used without an encoder, set open loop phi positive
+        // direction
+        writeToControllerOrDieTrying(motor, TMC4671_OPENLOOP_MODE, 0x00000000);
+        writeToControllerOrDieTrying(motor, TMC4671_PHI_E_SELECTION,
+                                     TMC4671_PHI_E_OPEN_LOOP);
+        writeToControllerOrDieTrying(motor, TMC4671_OPENLOOP_ACCELERATION, 0x0000003C);
+
+        // represents effective voltage applied to the motors (% voltage)
+        writeToControllerOrDieTrying(motor, TMC4671_UQ_UD_EXT, 0x00000799);
+
+        // uq_ud_ext mode
+        writeToControllerOrDieTrying(motor, TMC4671_MODE_RAMP_MODE_MOTION, 0x00000008);
+
+        // 10 RPM
+        writeToControllerOrDieTrying(motor, TMC4671_OPENLOOP_VELOCITY_TARGET, 0x0000000A);
+    }
+
+    for (int num_iterations = 0;
+         num_iterations < 10 &&
+         std::any_of(calibrated_motors.begin(), calibrated_motors.end(),
+                     [](bool calibration_status) { return !calibration_status; });
+         ++num_iterations)
+    {
+        for (uint8_t motor = 0; motor < NUM_DRIVE_MOTORS; ++motor)
+        {
+            if (calibrated_motors[motor])
+            {
+                continue;
+            }
+            // now read back the velocity
+            int read_back_velocity = tmc4671_readInt(motor, TMC4671_ABN_DECODER_COUNT);
+            LOG(INFO) << MOTOR_NAMES[motor] << " read back: " << read_back_velocity
+                      << " and initially read: " << initial_velocities[motor];
+
+            if (read_back_velocity != initial_velocities[motor])
+            {
+                calibrated_motors[motor] = true;
+            }
+        }
+
+        // sleep for 100 milliseconds
+        usleep(MICROSECONDS_PER_MILLISECOND * 100);
+    }
+
+    bool calibrated = true;
+    for (uint8_t motor = 0; motor < NUM_DRIVE_MOTORS; ++motor)
+    {
+        if (!calibrated_motors[motor])
+        {
+            calibrated = false;
+            LOG(WARNING) << "Encoder calibration check failure. " << MOTOR_NAMES[motor]
+                         << " did not change as expected";
+        }
+    }
+    if (!calibrated)
+    {
+        LOG(FATAL)
```

Comment:
nima: nit: This right now prints only the name of the first motor that has its encoder calibrated, then crashes. It might be better to combine all names in the same LOG(FATAL), or have them as individual WARNINGs, and then a single FATAL message.

Reply:
each motor is logged with a warning now and then a final fatal message @nimazareian